src/test/java/software/amazon/awssdk/crt/test/CredentialsProviderTest.java

```diff
@@ -151,4 +157,41 @@ public Credentials getCredentials() {
             fail(ex.getMessage());
         }
     }
+
+    @Test
+    public void testCreateDestroyProfile_ValidProfile() throws IOException {
+        Path confPath = Files.createTempFile("testCreateDestroyProfile_ValidProfile_conf_", "");
+        Path credsPath = Files.createTempFile("testCreateDestroyProfile_ValidProfile_creds_", "");
+        Files.write(credsPath, Arrays.asList("[default]"));
+
+        ProfileCredentialsProvider.Builder builder = ProfileCredentialsProvider
+                .builder()
+                .withConfigFileNameOverride(confPath.toString())
+                .withCredentialsFileNameOverride(credsPath.toString());
+
+        try (ProfileCredentialsProvider provider = builder.build()) {
+            assertNotNull(provider);
+            assertTrue(provider.getNativeHandle() != 0);
+        } finally {
+            Files.deleteIfExists(credsPath);
+        }
+    }
+
+    @Test(expected = CrtRuntimeException.class) // Because the profile files don't contain a [default] section
```

Comment:
A little confused here.  The lack of any information in the test file shouldn't afffect provider creation only credentials sourcing.

Reply:
The underlying C code reads the profile immediately upon creation of the provider. I believe it's looking for a role ARN to determine whether the provider will be STS-based or not. When it finds no default profile, it [throws an error](https://github.com/awslabs/aws-c-auth/blob/a5dbb0cb7fccb6405b16245f4be7cbd2225496bd/source/credentials_provider_profile.c#L392-L399) which gets wrapped as this CrtRuntimeException.

Reply:
Hmm, so it does